### PR TITLE
fix(javascript): correctly handle parentheses between await and optional chaining

### LINKED ISSUE #5857: Await Stripping Parentheses w/ Optional Chaining
**Prettier 1.16.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBDAzgTymABAMwFccYBLaXACTgBsaIB1CAJxoBMAKASl2AB0ouXADdUzXAA9cAXlwdUAd1SkYBCBAB0AIzHaaqABZcA-BoOkA3AIC+IADQgIABzLR0yUGOYQFABTEI7iiowhCkbPYgWsyoYADWcDAAyk6xpFAA5sgwzIRwDgYwALY0DObw6KlgcEmBKqTCKpjI4OjuDunocMwwvjEZRajI+Kg0XQ4AVugSAEIx8YlJqEVwADLpcMOj4yBTEknpGTRwAIqEEPBbY-kgqcxdzC06Wpj0UJFOzOkwDOEwBsgABwABgcnwgXQYMScLU+cAewk2DmYcAAjoRSCi+qgBkMkCNrg4ukVSNlcjd0IdjmcLpt8dsbjBUFpfmx-sgAEwOHLKGiHADCECKgxaUGgSJAhC6ABVmUECV1rNYgA)
```sh
--parser babylon
```

**Input:**
```js
async function HelloWorld() {
  var x = (await foo.bar.blah)?.hi;
}
```

**Output:**
```js
async function HelloWorld() {
  var x = await foo.bar.blah?.hi;
}

```

**Expected behavior:**
```js
async function HelloWorld() {
  var x = (await foo.bar.blah)?.hi;
}
```
This seems to be similar to https://github.com/prettier/prettier/issues/239 . Things work correctly with normal  property access instead of `?.`

### PR DESCRIPTION
**Pull request by evilebottnawi**
- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

fixes #5857

### CODE REVIEW

File: tests/optional_chaining/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -61,5 +65,9 @@ a?.b?.c.d?.e;
 (list || list2)?.length;
 (list || list2)?.[list || list2];
 
+async function HelloWorld() {
+  var x = await (foo.bar.blah)?.hi;
```
Comment (evilebottnawi) [inline tests/optional_chaining/__snapshots__/jsfmt.spec.js.snap]: Nit right places :trollface: WIP

**Comment by berrylaw** [comment]
Any partie

**Comment by evilebottnawi** [comment]
@berrylaw What do you mean?

**Comment by Sceat** [comment]
Can't wait for this 🤖 mergemergemergemerge

**Comment by berrylaw** [comment]
Same stage

「Sceat <notifications@github.com>」在 2019年4月27日 週六，上午9:20 寫道：

**Comment by evilebottnawi** [comment]
Feel free to feedback

**Comment by Sceat** [comment]
@evilebottnawi do u know about the process required to get it VsCode ? i guess it has to be published to npm and the extension updated by maintener ?

**Comment by evilebottnawi** [comment]
@Sceat I can't understand you

**Comment by Sceat** [comment]
@evilebottnawi just want to know the ETA for the prettier vscode extension with this fix

**Comment by j-f1** [comment]
We do releases approximately every 1.5 months, and I believe the VS Code extension uses whatever version you have installed, so once we release a new version, the VS Code extension will be updated.

**Pull request merged**